Mathlib/SetTheory/Game/PGame.lean

```diff
@@ -1902,6 +1910,46 @@ theorem insertRight_insertLeft {x x' x'' : PGame} :
   cases x; cases x'; cases x''
   dsimp [insertLeft, insertRight]
 
+/-! ### Removing an option -/
+
+/-- The PGame constructed by removing `'x` as a left option from `x`. -/
+def removeLeft (x : PGame.{u}) (i : x.LeftMoves) : PGame :=
+  match x with
+  | mk _ xr xL xR => mk ({ x // x ≠ i }) xr (fun k ↦ xL (k.val)) xR
```

Comment:
Is this definition really what you want? Remember that `PGame` equality is a very strong relation. Perhaps `¬ Identical x i` makes more sense?

Reply:
I might move this PR to `Game`: I wanted to have this for the replacement lemma, which operates under `PGame.setoid`.

Reply:
I think this is fine? `i` is an element in `x.LeftMoves`, it's not a PGame.
```suggestion
  | mk _ xr xL xR => mk { x // x ≠ i } xr (fun k ↦ xL k.val) xR
```

Reply:
The suggestion works (with an extra `val` cast)

> I think this is fine? `i` is an element in `x.LeftMoves`, it's not a PGame.

The main problem is that the strict equality means `removeLeft` is well defined for `PGame`, but for what I wanted, it wouldn't work well if I was strictly *trying* to remove an option. I think this PR is fine, but this is part of the many code duplication issues between `PGame` / a potential `IGame` / `Game`.

Reply:
[I've mentioned this duplication issue in Zulip](https://leanprover.zulipchat.com/#narrow/channel/116395-maths/topic/What.20to.20do.20about.20PGame/near/491135352).

Reply:
For that purpose it's probably appropriate to use equality (or inequality) in Game. If you remove all identical options, but some option with the same value remains, you still don't change the value of x.

Reply:
> For that purpose it's probably appropriate to use equality (or inequality) in Game. If you remove all identical options, but some option with the same value remains, you still don't change the value of x.

Absolutely. Though I do think there might be some value in keeping this definition for `PGame` and having a separate definition for `Game`. However, that decision entirely depends on how `PGame`/`IGame` refactoring goes.

Reply:
After some more Zulip conversation, it seems appropriate to keep this `PGame.removeLeft`/`PGame.removeRight`, and have corresponding remove APIs for `IGame` and `Game`.